src/pages/iou/steps/MoneyRequestAmountForm.js

```diff
@@ -0,0 +1,278 @@
+            return MoneyRequestUtils.stripCommaFromAmount(newAmountWithoutSpaces);
+        });
+    };
+
+    /**
+     * Update amount with number or Backspace pressed for BigNumberPad.
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit to enable Next button
+     *
+     * @param {String} key
+     */
+    const updateAmountNumberPad = useCallback(
+        (key) => {
+            if (shouldUpdateSelection && !textInput.current.isFocused()) {
+                textInput.current.focus();
+            }
+            // Backspace button is pressed
+            if (key === '<' || key === 'Backspace') {
+                if (currentAmount.length > 0) {
+                    const selectionStart = selection.start === selection.end ? selection.start - 1 : selection.start;
+                    const newAmount = `${currentAmount.substring(0, selectionStart)}${currentAmount.substring(selection.end)}`;
+                    setNewAmount(newAmount);
+                }
+                return;
+            }
+            const newAmount = MoneyRequestUtils.addLeadingZero(`${currentAmount.substring(0, selection.start)}${key}${currentAmount.substring(selection.end)}`);
+            setNewAmount(newAmount);
+        },
+        [currentAmount, selection, shouldUpdateSelection],
+    );
+
+    /**
+     * Update long press value, to remove items pressing on <
+     *
+     * @param {Boolean} value - Changed text from user input
+     */
+    const updateLongPressHandlerState = useCallback((value) => {
+        setShouldUpdateSelection(!value);
+        if (!value && !textInput.current.isFocused()) {
+            textInput.current.focus();
+        }
+    }, []);
+
+    /**
+     * Update amount on amount change
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit
+     *
+     * @param {String} text - Changed text from user input
+     */
+    const updateAmount = (text) => {
+        const newAmount = MoneyRequestUtils.addLeadingZero(MoneyRequestUtils.replaceAllDigits(text, fromLocaleDigit));
+        setNewAmount(newAmount);
+    };
+
+    /**
+     * Submit amount and navigate to a proper page
+     *
+     */
+    const submitAndNavigateToNextPage = useCallback(() => {
+        onSubmitButtonPress(currentAmount);
+    }, [onSubmitButtonPress, currentAmount]);
+
+    const formattedAmount = MoneyRequestUtils.replaceAllDigits(currentAmount, toLocaleDigit);
+    const buttonText = isEditing ? translate('common.save') : translate('common.next');
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            onEntryTransitionEnd={focusTextInput}
+        >
+            {({safeAreaPaddingBottomStyle}) => (
+                <View style={[styles.flex1, safeAreaPaddingBottomStyle]}>
+                    <HeaderWithBackButton
```

Comment:
`ScreenWrapper` and `HeaderWithBackButton` should belong to `NewRequestAmountPage`

Reply:
ok, so moving it out

Reply:
done! 🙂